src/edu/stevens/ece/TextEditor.java

```diff
@@ -26,15 +30,79 @@ public TextEditor() throws Exception {
 		editor = new JTextPane();
 		JScrollPane editorScrollPane = new JScrollPane(editor);
 		editor.setDocument(doc = new DefaultStyledDocument());
+		editor.setFont(defaultFont);
 		add(editorScrollPane, BorderLayout.CENTER);
+		addActions();
 		setVisible(true);
 	}
 	public void addActions() {
-		
+		new IrreversibleAction("SAVE") {
+			@Override public void doIt(ActionEvent e) {
+				save();
+			}
+		};
+		new IrreversibleAction("SAVEAS") {
+			@Override public void doIt(ActionEvent e) {
+				saveas();
+			}
+		};
+		new IrreversibleAction("OPEN") {
+			@Override public void doIt(ActionEvent e) {
+				open();
+			}
+		};
 	}
 	public void open() {
+		try {
+			fileChooser.setDialogTitle(lookupText("OPEN"));
+			int returnVal = fileChooser.showOpenDialog(this);
+			if (returnVal == JFileChooser.APPROVE_OPTION) {
+				File file = fileChooser.getSelectedFile();
+				Reader reader = new BufferedReader(new FileReader(file));
+				editorKit.read(reader, doc, 0);
+				reader.close();
+			}
+		} catch(Exception e) {
+			statusAlert(e);
+		}			
 	}
 	public void save() {
+		try {
+
+		} catch(Exception e) { // if anything goes wrong, alert in the status bar
+			statusAlert(e);
+		}
+	}
+	public void saveas() {
+		try {
+			fileChooser.setDialogTitle(lookupText("SAVEAS..."));
+			int returnVal = fileChooser.showOpenDialog(this);
+			if (returnVal == JFileChooser.APPROVE_OPTION) {
+				File file = fileChooser.getSelectedFile();
+				Writer writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(file)));
+				editorKit.write(writer, doc, 0, doc.getLength());
+				writer.close();
+			}
+		} catch(Exception e) { // if anything goes wrong, alert in the status bar
+			statusAlert(e);
+		}
+	}
+	public void openCompressed() {
```

Comment:
Great idea! No longer will need to fumble around to decompress, edit, and re-compress files. Would this allow modifying multiple files within a gzipped file. Would there be a risk of corrupting the GZip if the application is killed during a partial write?